internal/app/release.go

```diff
@@ -34,6 +34,20 @@ type release struct {
 	HelmFlags    []string          `yaml:"helmFlags"`
 	NoHooks      bool              `yaml:"noHooks"`
 	Timeout      int               `yaml:"timeout"`
+	Hooks        hooks             `yaml:"hooks"`
+}
+
+// hooks type which defines life-cycle hooks
+type hooks struct {
```

Comment:
I think this struct is fine for reading the data from the helmsman file but during the validation we could convert it into another one where we keep the "hook actions" in a `map[string]string` that way we can cleanup the code that needs to loop through them, what do you think?

Reply:
Thanks to the help of @zanderhavgaard , the struct is now replaced with a map.